iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/services/models/DeploymentServiceModel.java

```
@@ -23,55 +20,89 @@
         String deviceGroupName = StringUtils.EMPTY;
-        if (config.getLabels().containsKey(DEPLOYMENT_GROUP_NAME_LABEL)) {
-            deviceGroupName = config.getLabels().get(DEPLOYMENT_GROUP_NAME_LABEL);
+        if (deployment.getLabels().containsKey(ConfigurationsHelper.DEPLOYMENT_GROUP_NAME_LABEL)) {
+            deviceGroupName = deployment.getLabels().get(ConfigurationsHelper.DEPLOYMENT_GROUP_NAME_LABEL);
         }
         this.deviceGroup = new DeviceGroup(deviceGroupId, deviceGroupName, null);
 
         this.packageName = StringUtils.EMPTY;
-        if (config.getLabels().containsKey(DEPLOYMENT_PACKAGE_NAME_LABEL)) {
-            this.packageName = config.getLabels().get(DEPLOYMENT_PACKAGE_NAME_LABEL);
+        if (deployment.getLabels().containsKey(ConfigurationsHelper.DEPLOYMENT_PACKAGE_NAME_LABEL)) {
+            this.packageName = deployment.getLabels().get(ConfigurationsHelper.DEPLOYMENT_PACKAGE_NAME_LABEL);
         }
 
-        this.createdDateTimeUtc =  this.formatDateTimeToUTC(config.getCreatedTimeUtc());
-        this.priority = config.getPriority();
-        this.type = DeploymentType.edgeManifest;
-        this.deploymentMetrics = new DeploymentMetrics(config.getSystemMetrics(), config.getMetrics());
+        this.createdDateTimeUtc =  this.formatDateTimeToUTC(deployment.getCreatedTimeUtc());
+        this.priority = deployment.getPriority();
+
+        if (deployment.getLabels().containsKey(ConfigurationsHelper.PACKAGE_TYPE_LABEL) &&
+            !(StringUtils.isBlank(deployment.getLabels().get(ConfigurationsHelper.PACKAGE_TYPE_LABEL))))
+        {
+            if (deployment.getLabels().containsValue(PackageType.edgeManifest.toString()))
+            {
+                this.packageType = PackageType.edgeManifest;
+            }
+            else if (deployment.getLabels().containsValue(PackageType.deviceConfiguration.toString()))
+            {
+                this.packageType = PackageType.deviceConfiguration;
+            }
```

If it is neither maybe we should throw an exception?

---

added